File: base/src/main/java/com/rye/base/BaseFragment.java
package com.rye.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseFragment extends Fragment {
    private Unbinder unbinder;
    protected View mRoot;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        beforeInflateView();
        mRoot = inflater.inflate(getLayoutId(), container, false);
        return mRoot;
    }

    protected abstract @LayoutRes
    int getLayoutId();

    public void beforeInflateView() {
    }

    public void initWidget() {
    }

    public void initEvent() {
    }

    public View getView() {
        if (mRoot == null) {
            throw new IllegalArgumentException("根布局不能为空~");
        }
        return mRoot;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        unbinder = ButterKnife.bind(this, view);
        initWidget();
        initEvent();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }
}

File: base/src/main/java/com/rye/base/BaseFragment.kt
package com.rye.base

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.ViewGroup
import com.rye.base.mvp.PresenterHoler
import com.trello.rxlifecycle2.android.ActivityEvent
import io.reactivex.subjects.BehaviorSubject

/**
 *Created By RyeCatcher
 * at 2019/8/22
 */
abstract class BaseFragment: Fragment(),IView {
    protected  lateinit var TAG:String
    protected  lateinit var mContentView:ViewGroup
    protected  lateinit var presenterHelper:PresenterHoler
    /**
     * Rx lifecycle
     */
    private val lifecycleSubject=BehaviorSubject.create<ActivityEvent>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        TAG=javaClass.simpleName

        lifecycleSubject.onNext(ActivityEvent.CREATE)

        presenterHelper=PresenterHoler(this)

    }

}